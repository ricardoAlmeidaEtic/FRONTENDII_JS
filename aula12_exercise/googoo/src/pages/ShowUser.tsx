import { useState, useEffect } from 'react';

const ShowUser = () => {
  const [user, setUser] = useState("Ricardo");
  const [showUser, setShowUser] = useState(true);

  return (
    <div>
      <button onClick={() => setShowUser(!showUser)}>{showUser ? 'Esconder' : 'Mostrar'}</button>
      {showUser && (
        <>
          <div className="card">
            <h2>User: { user }</h2>
          </div>
        </>
      )}
    </div>
  )
}

export default ShowUser