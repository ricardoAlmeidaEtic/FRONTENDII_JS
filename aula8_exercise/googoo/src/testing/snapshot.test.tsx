import ShowUser from '../pages/ShowUser';
import { render } from '@testing-library/react';

test('Counter increments correctly', () => {
  const { container } = render(<ShowUser />);
  expect(container).toMatchSnapshot();
});