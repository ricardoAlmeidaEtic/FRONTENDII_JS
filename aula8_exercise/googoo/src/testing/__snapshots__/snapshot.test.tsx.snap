// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter increments correctly 1`] = `
<div>
  <div>
    <button>
      Esconder
    </button>
    <div
      class="card"
    >
      <h2>
        User: 
        Ricardo
      </h2>
    </div>
  </div>
</div>
`;
